core/src/commonMain/kotlin/com/powersync/attachments/storage/IOLocalStorageAdapter.kt

```diff
@@ -0,0 +1,114 @@
+package com.powersync.attachments.storage
+
+import com.powersync.attachments.LocalStorage
+import com.powersync.db.runWrappedSuspending
+import io.ktor.utils.io.core.readBytes
+import io.ktor.utils.io.core.remaining
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.withContext
+import kotlinx.io.Buffer
+import kotlinx.io.buffered
+import kotlinx.io.files.Path
+import kotlinx.io.files.SystemFileSystem
+
+/**
+ * Storage adapter for local storage using the KotlinX IO library
+ */
+public class IOLocalStorageAdapter : LocalStorage {
```

Comment:
I suppose for testing or if users want a custom file system, adding a `private val fileSystem: FileSystem = SystemFileSystem` and using that in the methods may be convenient?

Reply:
That's a fair recommendation. I've implemented this.